crates/vm/levm/src/opcode_handlers/environment.rs

```diff
@@ -172,6 +168,15 @@ impl VM {
         }
 
         let mut data = vec![0u8; size];
+        if calldata_offset > current_call_frame.calldata.len().into() {
+            memory::try_store_data(&mut current_call_frame.memory, dest_offset, &data)?;
+            return Ok(OpcodeSuccess::Continue);
+        }
+
+        let calldata_offset: usize = calldata_offset.try_into().map_err(|_err| {
+            VMError::Internal(crate::errors::InternalError::ArithmeticOperationOverflow)
+        })?;
```

Comment:
```suggestion
        let calldata_offset: usize = calldata_offset.try_into().map_err(|_err| {
            VMError::Internal(InternalError::ConversionError)
        })?;
```
This error is more accurate